File: CSharp/SpaceVIL/SpaceVIL/Items/Ellipse.cs
namespace SpaceVIL
{
    /// <summary>
    /// Ellipse is a subclass that extends from SpaceVIL.Primitive 
    /// for rendering an ellipse shape.
    /// </summary>
    public class Ellipse : Primitive
    {
        static int count = 0;
        /// <summary>
        /// Property to specify number of edges in an ellipse shape.
        /// <para/> Default: 16.
        /// </summary>
        public int Quality = 16;

        /// <summary>
        /// Default Ellipse constructor.
        /// </summary>
        public Ellipse() : base("Ellipse_" + count)
        {
            count++;
        }

        /// <summary>
        /// Constructs an Ellipse with specified number of edges in an ellipse shape.
        /// </summary>
        /// <param name="quality">Number of edges.</param>
        public Ellipse(int quality) : this()
        {
            this.Quality = quality;
        }

        /// <summary>
        /// Overridden method for stretching the ellipse shape relative to the current size. 
        /// Use in conjunction with GetTriangles() and SetTriangles() methods.
        /// <para/> Notice: This method is mainly for overriding only. SpaceVIL calls 
        /// this method if necessary and no need to call it manually.
        /// </summary>
        public override void MakeShape()
        {
            SetTriangles(GraphicsMathService.GetEllipse(GetWidth(), GetHeight(), 0, 0, Quality));
        }
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Ellipse.java
package com.spvessel.spacevil;

/**
 * Ellipse is a subclass that extends from com.spvessel.spacevil.Primitive for
 * rendering an ellipse shape.
 */
public class Ellipse extends Primitive {
    private static int count = 0;
    /**
     * Property to specify number of edges in an ellipse shape.
     * <p>
     * Default: 16.
     */
    public int quality = 16;

    /**
     * Default Ellipse constructor.
     */
    public Ellipse() {
        setItemName("Ellipse_" + count);
        count++;
    }

    /**
     * Constructs an Ellipse with specified number of edges in an ellipse shape.
     * 
     * @param quality Number of edges.
     */
    public Ellipse(int quality) {
        this();
        this.quality = quality;
    }

    /**
     * Overridden method for stretching the ellipse shape relative to the current
     * size. Use in conjunction with getTriangles() and setTriangles() methods.
     * <p>
     * Notice: This method is mainly for overriding only. SpaceVIL calls this method
     * if necessary and no need to call it manually.
     */
    @Override
    public void makeShape() {
        setTriangles(GraphicsMathService.getEllipse(getWidth(), getHeight(), 0, 0, quality));
    }
}